node/src/testing/test_block_builder.rs

```diff
@@ -0,0 +1,194 @@
+use std::collections::BTreeMap;
+
+use casper_hashing::Digest;
+use casper_types::{
+    system::auction::BLOCK_REWARD, EraId, ProtocolVersion, PublicKey, SecretKey, Timestamp, U512,
+};
+use rand::Rng;
+
+use crate::{
+    components::consensus::EraReport,
+    types::{Block, BlockHash, BlockPayload, Deploy, DeployHashWithApprovals, FinalizedBlock},
+};
+
+const MAX_ERA_FOR_RANDOM_BLOCK: u64 = 6;
+
+pub(crate) struct TestBlockBuilder {
+    era: Option<u64>,
+    height: Option<u64>,
+    protocol_version: ProtocolVersion,
+    is_switch: Option<bool>,
+    deploys: Vec<Deploy>,
+    state_root_hash: Option<Digest>,
+    parent_hash: Option<BlockHash>,
+    timestamp: Option<Timestamp>,
```

Comment:
Would be nice to order these in the same way as inside `Block` (and also for the functions below).